packages/nimble-components/src/table-column/date-text/index.ts

```diff
@@ -31,6 +124,141 @@ export class TableColumnDateText extends TableColumnTextBase {
             sortOperation: TableColumnSortOperation.basic
         };
     }
+
+    protected formatChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customLocaleMatcherChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customWeekdayChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customEraChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customYearChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customMonthChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customDayChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customHourChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customMinuteChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customSecondChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customTimeZoneNameChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customFormatMatcherChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customHour12Changed(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customTimeZoneChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customCalendarChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customDayPeriodChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customNumberingSystemChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customDateStyleChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customTimeStyleChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    protected customHourCycleChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    private updateColumnConfig(): void {
+        const columnConfig: TableColumnDateTextColumnConfig = {
+            formatter: this.createFormatter()
+        };
+        this.columnInternals.columnConfig = columnConfig;
+        this.validator.setCustomOptionsValidity(
+            columnConfig.formatter !== undefined
+        );
+    }
+
+    private createFormatter(): Intl.DateTimeFormat | undefined {
+        let options: Intl.DateTimeFormatOptions;
+        if (!this.format) {
+            options = {
+                dateStyle: 'medium',
+                timeStyle: 'medium'
+            };
+        } else {
+            options = this.getCustomFormattingOptions();
+        }
+        try {
+            return new Intl.DateTimeFormat(undefined, options);
```

Comment:
I've got a follow-on PR tee'd up that add support for setting the locale via `nimble-theme-provider`. In those changes, I update the tests to run under a specific locale ('en'). Since that should be along shortly, can we just wait for that?

Reply:
Are you saying we should not update the karma config or GitHub actions settings? I think we should anyway and should have tests that show it can be unconfigured and pickup the system locale.

Reply:
Added the locale flag to our karma config. I'll do the GitHub actions changes separately, as suggested.

Reply:
Created PR https://github.com/ni/nimble/pull/1409 for the GitHub actions changes.